scom/stackstate_checks/scom/scom.py

```diff
@@ -82,114 +52,136 @@ def sendAlerts(self, session, scom_ip, domain, username, password, component_id)
                     "time_added:%s" % event.get("timeadded")
                 ]
             })
+    def get_health_state(self,healthIconUrl):
+        if re.search('StatusOkComplete', healthIconUrl, flags=re.IGNORECASE) or re.search('healthy_ex', healthIconUrl,flags=re.IGNORECASE):
+           return 'Healthy'
+        elif re.search('StatusWarning', healthIconUrl, flags=re.IGNORECASE) or re.search('warning_ex', healthIconUrl,flags=re.IGNORECASE):
+           return 'Deviating'
+        elif re.search('StatusError', healthIconUrl, flags=re.IGNORECASE) or re.search('critical_ex', healthIconUrl,flags=re.IGNORECASE):
+           return 'Error'
+        elif re.search('StatusUnknownGreen', healthIconUrl, flags=re.IGNORECASE) or re.search('maintenance_ex',healthIconUrl,flags=re.IGNORECASE):
+           return 'Not Monitored'
+        else:
+           return str(healthIconUrl.split("/")[-1])
+
+    def get_type(self,session,component_id,domain, username, password, scom_ip):
+        if self.requests_counter > self.requests_threshold:
+           return
+        type_data = {
+                                "criteria": "(Id = '"+component_id+"')"
+                                }
+        type_response = (session.post('http://'+ scom_ip +'/OperationsManager/data/scomObjects',auth=HttpNtlmAuth(domain + '\\' + username, password),json=type_data["criteria"])).json()
+        self.requests_counter +=1
+        self.log.debug("Number of requets: "+str(self.requests_counter))
+        component_type = type_response.get("scopeDatas")[0].get("className").lower().replace(" ","-")
```

Comment:
```
type_response.get("scopeDatas")[0].get("className").lower().replace(" ","-")
```
This will fail if we get something unsuspected, can we add some guards around it

Reply:
the response is always something like
{
  "scopeDatas": [
    {
      "id": "e4750a18-1ae2-7a66-c54a-d0ab8caa1c85",
      "displayName": "Exchange Back End",
      "className": "IIS 10 Web Site",
      "path": "EX01.stackstate.lab",
      "fullName": "Microsoft.Windows.InternetInformationServices.10.0.WebSite:EX01.stackstate.lab;W3SVC/2"
    }
  ]
}
what i do is just some string manipulation to convert "IIS 10 Web Site" to "iis-10-web-site" and add it as the type of the component

how can we add some guards around it ? check if scopeDatas as well as className exists ?